astropy/units/astrophys.py

```diff
@@ -54,6 +54,15 @@
 def_unit(['solMass', 'M_sun', 'Msun'], _si.M_sun.value * si.kg, namespace=_ns,
          prefixes=True, doc="Solar mass",
          format={'latex': r'M_{\odot}', 'unicode': 'M⊙'})
+def_unit(['jupiterMass', 'M_jup', 'Mjup','M_jupiter', 'Mjupiter'],
+         _si.M_jup.value * si.kg, namespace=_ns,
+         prefixes=True, doc="Jupiter mass",
+         # LaTeX jupiter symbol requires wasysym
+         format={'latex': r'M_{Jup}', 'unicode': 'M♃'})
+def_unit(['earthMass', 'M_earth', 'Mearth'], _si.M_earth.value * si.kg, namespace=_ns,
+         prefixes=True, doc="Earth mass",
+         # LaTeX earth symbol requires wasysym
+         format={'latex': r'M_{\oplus}', 'unicode': 'M'})
```

Comment:
I think there's a glitch with the earth symbol here.  It looks like it's uf728 (which is undefined), but you probably meant u1f728 (ASTRONOMICAL SYMBOL FOR VERDIGRIS).  However, I'd recommend using u2295 (CIRCLED PLUS) instead, since I think it's a lot more widely available in various fonts.